pkg/segment/results/mresults/tests/metricsresults_test.go

```diff
@@ -404,3 +406,154 @@ func TestCalculateInterval(t *testing.T) {
 		}
 	}
 }
+
+func Test_GetResults_GTR(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetGreaterThan,
+				Constant:   1045,
+			},
+		})
+}
+
+func Test_GetResults_GTE(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			10800: 1045,
+		},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetGreaterThanOrEqualTo,
+				Constant:   100,
+			},
+		})
+}
```

Comment:
I'm confused reading these tests. I _think_ `10800` is comes from `test_GetResults_Ops()` downsampling to `3 * 3600` and because in that function you're making points from `i + 4 * 3600` so they'll downsample to `3 * 3600`, but it's not very clear and I'm not sure where `1045` is coming from.

Can you restructure your tests so it's easier to understand what they do and so it's clear that the values they should evaluate to are obviously the correct values?

Reply:
Yes, will do that